File: app/src/main/java/ru/mamykin/foreignbooksreader/views/DropboxView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.mamykin.foreignbooksreader.models.DropboxFile;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface DropboxView extends MvpView {
    void showFiles(List<DropboxFile> filesList);

    void hideFiles();

    void openBook(String path);

    void showLoadingItem(int position);

    void hideLoadingItem();

    void showLoading();

    void hideLoading();

    void showCurrentDir(String dir);

    void showAuth();

    void hideAuth();

    void showUpButton(boolean show);
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/views/DropboxView.kt
package ru.mamykin.foreignbooksreader.views

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.mamykin.foreignbooksreader.models.DropboxFile

@StateStrategyType(AddToEndSingleStrategy::class)
interface DropboxView : MvpView {
    fun showFiles(filesList: List<DropboxFile>)

    fun hideFiles()

    fun openBook(path: String)

    fun showLoadingItem(position: Int)

    fun hideLoadingItem()

    fun showLoading()

    fun hideLoading()

    fun showCurrentDir(dir: String)

    fun showAuth()

    fun hideAuth()

    fun showUpButton(show: Boolean)
}